File: frontend/src/services/productService.js
import api from './api';

export const productService = {
  // Backend: GET /api/products/getAllprod
  getAllProducts: async () => {
    const response = await api.get('/api/products/getAllprod');
    return response.data;
  },

  // Backend: GET /api/products/{id}
  getProductById: async (id) => {
    const response = await api.get(`/api/products/${id}`);
    return response.data;
  },

  // Backend: GET /api/products?search={name}
  searchProducts: async (name) => {
    const response = await api.get('/api/products', { params: { search: name } });
    return response.data;
  },

  // Backend: GET /api/products?category={categoryName}
  getProductsByCategory: async (categoryName) => {
    const response = await api.get('/api/products', { params: { category: categoryName } });
    return response.data;
  },

  // Backend: GET /api/products?minPrice={minPrice}&maxPrice={maxPrice}
  getProductsByPriceRange: async (minPrice, maxPrice) => {
    const response = await api.get('/api/products', { params: { minPrice, maxPrice } });
    return response.data;
  },

  // Backend: GET /api/products/paged?page=&size=&sortBy=&direction=
  getPagedProducts: async (page = 0, size = 10, sortBy = 'id', direction = 'asc') => {
    const response = await api.get('/api/products/paged', { params: { page, size, sortBy, direction } });
    return response.data;
  },

  // Backend: POST /api/products/addproduct — body is Product entity
  createProduct: async (productData) => {
    const response = await api.post('/api/products/addproduct', productData);
    return response.data;
  },

  // Backend: PUT /api/products/{id} — body is Product entity
  updateProduct: async (id, productData) => {
    const response = await api.put(`/api/products/${id}`, productData);
    return response.data;
  },

  // Backend: DELETE /api/products/deleteProduct/{productId}
  deleteProduct: async (id) => {
    const response = await api.delete(`/api/products/deleteProduct/${id}`);
    return response.data;
  },
};

File: src/main/java/com/dev/e_commorce/org/service/ProductService.java
package com.dev.e_commorce.org.service;

import java.util.List;


import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.Sort;
import org.springframework.stereotype.Service;

import com.dev.e_commorce.org.entity.Product;
import com.dev.e_commorce.org.repo.ProductRepository;

@Service
@org.springframework.transaction.annotation.Transactional
public class ProductService {

	@Autowired
	private ProductRepository productRepository;
	
	public Product addProduct(Product newprod) {
		return productRepository.save(newprod);
	}
	
	public Product getProductById(Long id) {
		return productRepository.findById(id)
				.orElseThrow(() -> new RuntimeException("Product not found with id: " + id));
	}

	public Product updateProduct(Long id, Product updatedprod) {
		Product oldprod = productRepository.findById(id).orElseThrow(() -> new RuntimeException("Product not found !"));
		oldprod.setName(updatedprod.getName());
		oldprod.setDescription(updatedprod.getDescription());
		oldprod.setPrice(updatedprod.getPrice());
		oldprod.setStockQuantity(updatedprod.getStockQuantity());
		return productRepository.save(oldprod);	
	}
	
	public void deleteProduct(Long id) {
		Product prod = getProductById(id);
		productRepository.delete(prod);
	}
	
	public List<Product> getAllProd(){
		return productRepository.findAll();
	}
	
	public List<Product> getBySeller(Long sellerId){
		return productRepository.findBySellerId(sellerId);
	}
	
	public List<Product> searchProduct(String keyword){
		return productRepository.findByNameContainingIgnoreCase(keyword);
	}
	public List<Product> getProductByCategory(String category){
		return productRepository.findByCategoryNameIgnoreCase(category);
	}
	public List<Product> getProductByPriceRange(Double minprice, Double maxprice){
		return productRepository.findByPriceBetween(minprice, maxprice);
	}
    public Page<Product> getPagedProducts(int page, int size, String sortBy, String direction) {
        Sort sort = direction.equalsIgnoreCase("asc") ? Sort.by(sortBy).ascending() : Sort.by(sortBy).descending();
        Pageable pageable = PageRequest.of(page, size, sort);
        return productRepository.findAll(pageable);
    }

}
